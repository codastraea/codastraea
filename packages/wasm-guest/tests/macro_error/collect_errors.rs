use serpent_automation_wasm_guest::workflow;

pub struct X;

impl X {
    #[workflow]
    fn counter<'a, const X: usize, T>(&self, x: usize) {}
}

fn main() {}
