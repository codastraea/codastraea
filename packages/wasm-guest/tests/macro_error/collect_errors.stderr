error: `workflow` functions should have no runtime parameters
 --> tests/macro_error/collect_errors.rs:7:39
  |
7 |     fn counter<'a, const X: usize, T>(&self, x: usize) {}
  |                                       ^^^^^^^^^^^^^^^

error: `workflow` functions should have no type parameters
 --> tests/macro_error/collect_errors.rs:7:15
  |
7 |     fn counter<'a, const X: usize, T>(&self, x: usize) {}
  |               ^^^^^^^^^^^^^^^^^^^^^^^

error: `workflow` functions should have no lifetime parameters
 --> tests/macro_error/collect_errors.rs:7:15
  |
7 |     fn counter<'a, const X: usize, T>(&self, x: usize) {}
  |               ^^^^^^^^^^^^^^^^^^^^^^^

error: `workflow` functions should have no `const` parameters
 --> tests/macro_error/collect_errors.rs:7:15
  |
7 |     fn counter<'a, const X: usize, T>(&self, x: usize) {}
  |               ^^^^^^^^^^^^^^^^^^^^^^^

error: `workflow` functions should be async
 --> tests/macro_error/collect_errors.rs:7:5
  |
7 |     fn counter<'a, const X: usize, T>(&self, x: usize) {}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `workflow` functions should be free functions, not methods
 --> tests/macro_error/collect_errors.rs:7:39
  |
7 |     fn counter<'a, const X: usize, T>(&self, x: usize) {}
  |                                       ^^^^^
